src/operator/tensor/sparse_retain.cc

```diff
@@ -0,0 +1,53 @@
+/*!
+ * Copyright (c) 2017 by Contributors
+ * \file sparse_retain.cc
+ * \brief
+*/
+
+#include "./sparse_retain-inl.h"
+namespace mxnet {
+namespace op {
+
+NNVM_REGISTER_OP(sparse_retain)
+.describe(R"code(pick rows specified by user input index array from a row sparse matrix
+and save them in the output sparse matrix.
+
+Example::
+
+  data = [[1, 2], [3, 4], [5, 6]]
+  indices = [0, 1, 3]
+  shape = (4, 2)
+  rsp_in = row_sparse(data, indices)
```

Comment:
Could you make this example more complete? e.g. `mx.nd.row_sparse(...)` so user can copy paste to run?